src/mqtt_client/mod.rs

```diff
@@ -141,8 +170,10 @@ where
     mqtt: minimq::Minimq<Stack, Clock, MESSAGE_SIZE, 1>,
     settings: Settings,
     state: sm::StateMachine<sm::Context<Clock, Settings>>,
-    settings_prefix: String<MAX_TOPIC_LENGTH>,
     prefix: String<MAX_TOPIC_LENGTH>,
+    listing_state: Option<MiniconfIter<Settings>>,
+    properties_cache: Option<Vec<u8, MESSAGE_SIZE>>,
```

Comment:
I suspect a single Option for both both doesn't work, right?

```
pending_response: Option<(Vec<u8, MESSAGE_SIZE>, Response<32>)>,
```

Reply:
The properties cache has to be used for both listing and the pending response, so it belongs to more than just one cached thing, hence why it got broken out.

Reply:
Ack